Autopilot/Inc/SensorFusion.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * Sensor Fusion - Converts raw sensor data into human readable formatted structs
+ * Author: Lucy Gong, Dhruv Rawat, Anthony Bertnyk
+ */
+
+#include "AttitudeDatatypes.hpp"
+
+#ifndef SENSORFUSION_HPP
+#define SENSORFUSION_HPP
+
+// -1 = FAILED
+// 0 = SUCCESS
+// 1 = Old Data
+struct SFError_t{
+    int errorCode;
+};
+
+struct SFOutput_t {
+    float roll, pitch, yaw; //rad
+    float rollRate, pitchRate, yawRate; //rad/s
+    float airspeed; //m/s
+    float altitude; //m
+    float altitudeSpeed; //m/s
+    long double latitude; //Decimal degrees
+    float latitudeSpeed; //m/s
+    long double longitude; //Decimal degrees
+    float longitudeSpeed; //m/s
+};
+
+/**
+ * Initiate sensor fusion.
+ */ 
+void SF_Init(void);
+
+/**
+ * Get fused sensor data.
+ * @param[out]    output  Reference to an output struct for fused data.
+ */ 
+SFError_t SF_GetResult(SFOutput_t *output);
```

Comment:
I feel like the following should be added: (Feel free to disagree)

- A static instance of the SFOutput_t struct that will be populated with the latest sensor fusion calculations (maybe this should be placed in the .cpp file...)
- A new function called SF_GenerateNewResults(), or some other name, that will be called by the RTOS. This will collect sensor data (maybe by using helper functions) and generate new SFOutput_t data

Now, SF_GetResult() will only return the current values stored in the static SFOutput_t struct declared either in this file or the .cpp file.

Reply:
yes, static instance should live in the .cpp